staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1443,3 +1541,45 @@ func sliceTypeAssertion(original, patch interface{}) ([]interface{}, []interface
 	}
 	return typedOriginal, typedPatch, nil
 }
+
+// extractReplaceKeysPatchStrategy process patch strategy, which is a string may contains multiple
+// patch strategies seperated by ",". It returns a boolean var indicating if it has
+// replaceKeys strategies and a string or other strategy.
+func extractReplaceKeysPatchStrategy(strategies []string) (bool, string, error) {
+	switch len(strategies) {
+	case 0:
+		return false, "", nil
+	case 1:
+		singleStrategy := strategies[0]
+		switch singleStrategy {
+		case replaceKeysStrategy:
+			return true, "", nil
+		default:
+			return false, singleStrategy, nil
+		}
+	case 2:
+		switch {
+		case strategies[0] == replaceKeysStrategy:
+			return true, strategies[1], nil
+		case strategies[1] == replaceKeysStrategy:
+			return true, strategies[0], nil
+		default:
+			return false, "", fmt.Errorf("unexpected patch strategy: %v\n", strategies)
+		}
+	default:
+		return false, "", fmt.Errorf("unexpected patch strategy: %v\n", strategies)
```

Comment:
I don't think you want to include a newline in the value of the error object you're returning here.